model/action.go

```diff
@@ -56,34 +61,54 @@ func (a *Action) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-// FunctionRef ...
+// FunctionRef defines the reference to a reusable function definition
 type FunctionRef struct {
 	// Name of the referenced function
 	RefName string `json:"refName" validate:"required"`
 	// Function arguments
+	// TODO: validate it as required if function type is graphql
 	Arguments map[string]interface{} `json:"arguments,omitempty"`
 	// String containing a valid GraphQL selection set
+	// TODO: validate it as required if function type is graphql
 	SelectionSet string `json:"selectionSet,omitempty"`
+
+	// Invoke specifies if the subflow should be invoked sync or async.
+	// Defaults to sync.
+	Invoke string `json:"invoke,omitempty" validate:"required,oneof=async sync"`
```

Comment:
isn't better to have something like this:
` Invoke InvokeKind `json:"invoke,omitempty" validate:"omitempty,min=1"`

Reply:
I'd like to define `InvokeKind` enum for this value, and it's will be fixed in another PR (it seems appear in other struct too).

If we use validate with min=1, we must add a custom validate function for this struct (at this moment, the validator's oneof tag satisfy our requirement, so i use it).